src/core/gui/XournalppCursor.cpp

```diff
@@ -515,6 +520,68 @@ auto XournalppCursor::createHighlighterOrPenCursor(int size, double alpha) -> Gd
 }
 
 
+auto XournalppCursor::createHorizontalLineCursor() -> GdkCursor* {
+    GtkWidget* theWidget = control->getWindow()->getXournal()->getWidget();
+    GdkWindow* theWindow = gtk_widget_get_window(theWidget);
+
+    gint x, y;
+    GdkDevice *mouse_device;
+
+    #if GTK_CHECK_VERSION (3,20,0)
+    GdkSeat *seat = gdk_display_get_default_seat (gdk_display_get_default ());
+    mouse_device = gdk_seat_get_pointer (seat);
+    #else
+    GdkDeviceManager *devman = gdk_display_get_device_manager (gdk_display_get_default ());
+    mouse_device = gdk_device_manager_get_client_pointer (devman);
+    #endif
+
+    gdk_window_get_device_position (theWindow, mouse_device, &x, &y, nullptr);
+
+    Layout* const layout = this->control->getWindow()->getLayout();
+    XojPageView* page = layout->getPageViewAt(x, y);
+    if (page) {
+        Range range = page->getVisiblePart();
+        double zoom = control->getZoomControl()->getZoom();
+        int width = static_cast<int>(std::round(range.getWidth() * zoom));
+        int height = 5; // Five is the default cursor size used elsewhere in the app.
+        gint dx = x - static_cast<int>(range.getX() * zoom) - page->getX();
+        if (this->currentCursorFlavour == (dx << (sizeof(gint) * 4) | width)) {
+            return nullptr;
+        }
+
+        cairo_surface_t* crCursor = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, width, height);
```

Comment:
nit: use the cairo smart pointers in `util/raii/CairoWrappers.h` to avoid the manual memory management.

Reply:
Interesting. Do I need to call `surface.release()` instead of `cairo_surface_destroy(surface)` when I'm done with it, or can I just let it go out of scope and let the stack deal with it?

Reply:
Also, I guess this is something we should update with the rest of this file at some point, because a lot of these functions use surfaces and the like.

Reply:
> Interesting. Do I need to call `surface.release()` instead of `cairo_surface_destroy(surface)` when I'm done with it, or can I just let it go out of scope and let the stack deal with it?

Just let the stack do its deed